docs/add-a-new-datasource.md

```diff
@@ -0,0 +1,70 @@
+
+Please use the `skaff` tool to generate new datasource and test templates for any new resource. Doing so will ensure that any boilerplate code, structural best practices and repetitive naming is done for you and always represents our most current standards.
+
+## Prerequisites
+
+If this is the first addition of a resource or datasource for a new service, please ensure the Service Client for the new service has been added and merged. See [Adding a new Service](add-a-new-service.md) for details.
+
+## Steps to Add a Datasource
+
+### Fork the provider and create a feature branch
+
+For a new resources use a branch named `f-{datasource name}` for example: `f-ec2-vpc`. See [Raising a Pull Request](raising-a-pull-request.md) for more details.
+
+### Name the datasource
+
+Either by creating the file manually, or using `skaff` to generate a template.
+
+All resources should be named with the following pattern: `aws_<service>_<name>`
+
+Where `<service>` is the AWS short service name that matches the key in the `serviceData` map in the `conns` package (created via the [Adding a new Service](add-a-new-service.md) )
+
+Where `<name>` represents the conceptual infrastructure represented by the create, read, update, and delete methods of the service API. It should be a singular noun. For example, in an API that has methods such as `CreateThing`, `DeleteThing`, `DescribeThing`, and `ModifyThing` the name of the resource would end in `_thing`.
+
+### Fill out the Datasource Schema
+
+In the `internal/service/<service>/<service>.go` file you will see a `Schema` property which exists as a map of `Schema` objects. This relates the AWS API data model with the Terraform resource itself. For each property you want to make available in Terraform, you will need to add it as an attribute, and choose the correct data type.
+
+Attribute names are to specified in `camel_case` as opposed to the AWS API which is `CamelCase`
+
+### Implement Read Handler
+
+These will map the AWS API response to the datasource schema. You will also need to handle different response types (including errors correctly). For complex attributes you will need to implement Flattener or Expander functions. The [Data Handling and Conversion Guide](data-handling-and-conversion.md) covers everything you need to know for mapping AWS API responses to Terraform State and vice-versa. The [Error Handling Guide](error-handling.md) covers everything you need to know about handling AWS API responses consistently.
+
+#### Write passing Acceptance Tests
+In order to adequately test the resource we will need to write a complete set of Acceptance Tests. You will need an AWS account for this which allows the creation of that resource. See [Writing Acceptance Tests](running-and-writing-acceptance-tests.md) for a detailed guide on how to approach these.
```

Comment:
It seems like we are using the words resource and datasource interchangeably, which can be confusing if I am brand new and trying to keep a the idea of a `resource` separate from a `datasource`